store/expiration_store.go

```diff
@@ -48,8 +49,20 @@ func NewExpirationStore[K comparable, V any](ctx context.Context, store Store[K,
 	return s
 }
 
+// Set associates the given value with the given key and sets the expiration time.
```

Comment:
this comment looks wrong - looks like just a copy-paste of Set's comment down below